siliconcompiler/schema/schema_cfg.py

```diff
@@ -3507,96 +3595,185 @@ def schema_asic(cfg):
-            switch="-constraint_check 'scenario <str>'",
-            example=["cli: -constraint_check 'worst check setup'",
-                    "api: chip.add('constraint','worst','check','setup')"],
+            shorthelp="Constraint: timing checks",
+            switch="-constraint_timing_check 'scenario <str>'",
+            example=[
+                "cli: -constraint_timing_check 'worst check setup'",
+                "api: chip.add('constraint', 'timing', 'worst','check','setup')"],
             schelp="""
             List of checks for to perform for the scenario. The checks must
             align with the capabilities of the EDA tools and flow being used.
             Checks generally include objectives like meeting setup and hold goals
             and minimize power. Standard check names include setup, hold, power,
             noise, reliability.""")
 
+    # COMPONENTS
+    scparam(cfg, ['constraint', 'component', instance, 'outline'],
+            sctype='[(float,float)]',
+            shorthelp="Constraint: Component outline",
+            switch="-constraint_component_outline 'i0 <(float,float)'>",
+            example=[
+                "cli: -constraint_component_outline 'i0 (0,0)'",
+                "api: chip.add('constraint', 'component', 'i0', 'outline', (0,0))"],
+            schelp="""
+            List of (x,y) points that constrains the outline of a named
+            instance. Simple rectangle areas can be defined
+            with two points, one for the lower left corner and one for the
+            upper right corner. The 'outline' parameter is a goal, not an
+            exact specification. The compiler and component system may adjust
+            sizes to meet competing goals such as manufacturing design
+            rules.""")
+
+    scparam(cfg, ['constraint', 'component', instance, 'placement'],
+            sctype='(float,float,float)',
+            shorthelp="Constraint: Component placement",
+            switch="-constraint_component_placement 'inst <(float,float, float)>'",
+            example=[
+                "cli: -constraint_component_placement 'i0 (2.0,3.0,0.0)'",
+                "api: chip.set('constraint', 'component', 'i0', 'placement', (2.0,3.0,0.0)"],
+            schelp="""
+            Placement location of a named component, specified as a (x,y,z) tuple of
+            floats. The location refers to the placement of the center/centroid of the
+            component. The 'placement' parameter is a goal/intent, not an exact specification.
+            The compiler and layout system may adjust sizes to meet competing
+            goals such as manufacturing design  rules and grid placement
+            guidelines. The 'z' coordinate shall be set to 0 for planar systems
+            with only (x,y) coordinates. Discretized systems like PCB stacks,
+            package stacks, and breadboards only allow a reduced
+            set of floating point values (0,1,2,3). The user specifying the
```

Comment:
This isn't a big deal, but we may need to account for floating-point imprecision if we use the Z-values as discrete steps. We should be okay if we use round numbers and don't perform arithmetic on them, but for example:

```
>>> a = 1.6
>>> '%.20f'%a
'1.60000000000000008882'
```